avocado/utils/configure_network.py

```diff
@@ -10,193 +10,388 @@
         """
-        Set MTU size in peer interface
+        cmd = "cat /sys/class/net/{}/address".format(self.name)
+        try:
+            return _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to get hw address: {}".format(ex))
+
+    def get_link_state(self):
+        """Method used to get the current link state of this interface.
+
+        This method will return 'up', 'down' or 'unknown', based on the
+        network interface state. Or it will raise a NWException if is
+        unable to get the interface state.
         """
-        cmd = "ip link set %s mtu %s" % (peer_interface, mtu)
+        cmd = "cat /sys/class/net/{}/operstate".format(self.name)
         try:
-            self.session.cmd(cmd)
-            cmd = "ip add show %s" % peer_interface
-            mtuvalue = self.session.cmd(cmd).stdout.decode("utf-8").split()[4]
-            if mtuvalue == mtu:
-                return True
-        except Exception as ex:  # pylint: disable=W0703
-            log.error("setting MTU value in peer failed: %s", ex)
-        return False
-
-    def get_peer_interface(self, peer_ip):
+            return _run_command(cmd, self.remote_session)
+        except process.CmdError as e:
+            msg = ('Failed to get link state. Maybe the interface is '
+                   'missing. {}'.format(e))
+            raise NWException(msg)
+
+    def get_mtu(self):
+        """Return the current MTU value of this interface.
+
+        This method will try to get the current MTU value, if fails will
+        raise a NWException.
         """
-        get peer interface from peer ip
+        try:
+            return self._get_interface_details()[0].get('mtu')
+        except (KeyError, IndexError):
+            raise NWException("Could not get MUT value.")
+
+    def ping_check(self, peer_ip, options=None):
+        """This method will try to ping a peer address (IPv4 or IPv6).
+
+        You should provide a IPv4 or IPV6 that would like to ping. This
+        method will try to ping the peer and if fails it will raise a
+        NWException.
+
+        :param peer_ip: Peer IP address (IPv4 or IPv6)
+        :param options: ping command options. Default is None
         """
-        cmd = "ip addr show"
+        cmd = "ping -I {} {}".format(self.name, peer_ip)
+        if options is not None:
+            cmd = "{} {}".format(cmd, options)
         try:
-            for line in self.session.cmd(cmd).stdout.decode("utf-8") \
-                                                    .splitlines():
-                if peer_ip in line:
-                    peer_interface = line.split()[-1]
-        except Exception as ex:  # pylint: disable=W0703
-            if peer_interface == "":
-                log.error("unable to get peer interface: %s", ex)
+            _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to ping: {}".format(ex))
```

Comment:
in ping_check() aslo need return value True or False for checking in our script.

Reply:
Same here.